### PR TITLE
Cleaner npm scripts with standard linting built in ☘

### PR DESCRIPTION
**Pull request by JaKXz**
Here's an easy one @dtinth - this is my suggestion for organizing npm scripts, making them easier to read and follow:
- [standard] linting in `pretest`
- tests must pass before publishing
- cross-platform friendly! :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,11 +4,14 @@
   "description": "Babel 6.x plugin to add instrument code with Istanbul-compatible `__coverage__` variable.",
   "main": "lib/index.js",
   "scripts": {
-    "build": "NODE_ENV=development ./node_modules/.bin/babel src -d lib",
-    "clean": "rm -rf lib; rm -rf lib-cov",
-    "mocha": "env NODE_ENV=test BABEL_PLUGIN__COVERAGE__TEST=1 nyc -r lcov -r text mocha",
-    "prepublish": "npm run clean && npm run build",
-    "test": "npm run clean && npm run build && env BABEL_DISABLE_CACHE=1 ./node_modules/.bin/babel src --plugins $(pwd)/lib/index -d lib-cov && npm run mocha"
+    "prebuild": "npm run clean",
+    "build": "NODE_ENV=development babel src -d lib",
```
Comment (dtinth) [inline package.json]: Does this one needs `cross-env` as well?

File: package.json [hunk 2]
```diff
@@ -4,11 +4,14 @@
...
+    "clean": "rimraf lib/ lib-cov/",
+    "premocha": "cross-env BABEL_DISABLE_CACHE=1 babel src --plugins $(pwd)/lib/index -d lib-cov",
+    "mocha": "cross-env NODE_ENV=test BABEL_PLUGIN__COVERAGE__TEST=1 nyc -r lcov -r text mocha",
+    "prepublish": "npm test",
+    "pretest": "standard src/** test.js && npm run build",
```
Comment (dtinth) [inline package.json]: Maybe extract the `standard` command to `npm run lint`?

**Comment by dtinth** [comment]
Hello! Thanks for your contribution. 👍  I’ve written some comments (above) 😄.

**Comment by JaKXz** [comment]
Thanks for the feedback, and good catches! :) I also took the liberty of adding an [`.editorconfig`](http://editorconfig.org/) file for the sake of future contributions!

**Comment by dtinth** [comment]
👍 👍 👍 

Thank you! This sure will help a lot for future contributors. 🎉

**Pull request merged**